qiskit/algorithms/optimizers/steppable_optimizer.py

```diff
@@ -0,0 +1,213 @@
+from abc import abstractmethod, ABC
+from dataclasses import dataclass, field
+from typing import Dict, Any, Union, Callable, Optional, Tuple, List
+
+from .optimizer import Optimizer, POINT, OptimizerResult
+import numpy as np
+
+CALLBACK = Callable[[int, np.ndarray, float, float], None]
+
+
+@dataclass
+class AskObject(ABC):
+    """Base class for return type of method SteppableOptimizer.ask()"""
```

Comment:
Could you add class field documentations that state what `x_fun` and `x_jac` are?